From patchwork Wed Nov 14 22:24:15 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Jonathan Marek <jonathan@marek.ca>
X-Patchwork-Id: 10683199
Return-Path: <linux-arm-msm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 7804414DB
	for <patchwork-linux-arm-msm@patchwork.kernel.org>;
 Wed, 14 Nov 2018 22:28:30 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 6964727F85
	for <patchwork-linux-arm-msm@patchwork.kernel.org>;
 Wed, 14 Nov 2018 22:28:30 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 5C6AE2BE9E; Wed, 14 Nov 2018 22:28:30 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id F1CF427F85
	for <patchwork-linux-arm-msm@patchwork.kernel.org>;
 Wed, 14 Nov 2018 22:28:29 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S2388427AbeKOIdd (ORCPT
        <rfc822;patchwork-linux-arm-msm@patchwork.kernel.org>);
        Thu, 15 Nov 2018 03:33:33 -0500
Received: from mail-qk1-f195.google.com ([209.85.222.195]:35094 "EHLO
        mail-qk1-f195.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S2388800AbeKOIdV (ORCPT
        <rfc822;linux-arm-msm@vger.kernel.org>);
        Thu, 15 Nov 2018 03:33:21 -0500
Received: by mail-qk1-f195.google.com with SMTP id w204so28797418qka.2
        for <linux-arm-msm@vger.kernel.org>;
 Wed, 14 Nov 2018 14:28:15 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=marek-ca.20150623.gappssmtp.com; s=20150623;
        h=from:to:cc:subject:date:message-id:in-reply-to:references;
        bh=iDpxwhz52J7768fW56aWS5n/NR5UxFIVBUXbmrpXRCo=;
        b=tq6OKscINrxkx2a6Qn/Sx1fSizwV3K4knjNbLkL/CjHYLvQkbici2eRFh2jfngOIGW
         BcMMEHZUklELL7iCdACTXxHzgJ8u3YDhA16CJXy5HHH7pTsHbvQTfyhJp6GvFV+eeA/i
         JvrskShm+sFTPnWmFyNTRcHi5zk6hZ753UE/+/bQx/A2/nyHulTHdbbxKB2snn2QuNSh
         UavKjkbIaRuyCp7Gzbl9ngK9933Ir+xswr/gpLuFikrjHqv7zooiRi7A2/mSsKjsNu/X
         wiJenoBQe4X1HhIOY3ZIVgonE+6oV+0KU9DQzXKbULBj7OK2bikd+C757CiRn1Jt0Pmc
         ow1g==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references;
        bh=iDpxwhz52J7768fW56aWS5n/NR5UxFIVBUXbmrpXRCo=;
        b=jAN0EvxYdOuxdZOvu+jsE4cTnL34+MGEs+Gngpgb2h+rerMoCsEt2ni+fQQjvnloX4
         NS27KZPVmqZXPJ0W4rlA1TuBSW3nnMH9rK62Z0zIuMoi83TI94BDhEfHfEMFBiBki6PE
         K1RnrBFuRt2KRzM9aaQGJrBvB80RPWpY9q1Oj3zmx5rYGXf7pzms0vFF/krz63f/4ykE
         pfjMSdlyHjJZHnZSOy6qNS8lGpQl7cXKx977KNPSlrkRGz5cTigRpBhhRN2FfA6U6ueg
         n2jQzkuWLV4NeJT8iQXvm6kU+VngJWkK+f24ISjKgEMkZ+sK/jmTU/DmBqQWqvJP0p6J
         xdbA==
X-Gm-Message-State: AGRZ1gLGXR3dgcRoXXdiTYG2VvMk4tzc4vMLvCK5JWGi2xWuIo5fw/HB
        ADYoct3Laytr7DlZ8GfHU5Yzvw==
X-Google-Smtp-Source: 
 AJdET5cYiesR/E6cnc8bQ69BrUlD4RuhoRc8yv/zIkPOIhY0wOdYyhiDxyi7qAUXygN0mMZ/twegtA==
X-Received: by 2002:aed:34e6:: with SMTP id x93mr3785103qtd.156.1542234494751;
        Wed, 14 Nov 2018 14:28:14 -0800 (PST)
Received: from localhost.localdomain (modemcable014.247-57-74.mc.videotron.ca.
 [74.57.247.14])
        by smtp.gmail.com with ESMTPSA id
 24sm24067595qkp.65.2018.11.14.14.28.13
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Wed, 14 Nov 2018 14:28:14 -0800 (PST)
From: Jonathan Marek <jonathan@marek.ca>
To: freedreno@lists.freedesktop.org
Cc: Rob Clark <robdclark@gmail.com>, David Airlie <airlied@linux.ie>,
        Sean Paul <seanpaul@chromium.org>,
        linux-arm-msm@vger.kernel.org (open list:DRM DRIVER FOR MSM ADRENO
 GPU),
        dri-devel@lists.freedesktop.org (open list:DRM DRIVER FOR MSM ADRENO
        GPU), linux-kernel@vger.kernel.org (open list)
Subject: [PATCH 8/9] drm/msm/mdp5: add config for msm8917
Date: Wed, 14 Nov 2018 17:24:15 -0500
Message-Id: <20181114222418.15960-8-jonathan@marek.ca>
X-Mailer: git-send-email 2.17.1
In-Reply-To: <20181114222418.15960-1-jonathan@marek.ca>
References: <20181114222418.15960-1-jonathan@marek.ca>
Sender: linux-arm-msm-owner@vger.kernel.org
Precedence: bulk
List-ID: <linux-arm-msm.vger.kernel.org>
X-Mailing-List: linux-arm-msm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

Add the mdp5_cfg_hw entry for MDP5 version v1.15 found on msm8917.

Signed-off-by: Jonathan Marek <jonathan@marek.ca>
---
 drivers/gpu/drm/msm/disp/mdp5/mdp5_cfg.c | 86 ++++++++++++++++++++++++
 1 file changed, 86 insertions(+)

diff --git a/drivers/gpu/drm/msm/disp/mdp5/mdp5_cfg.c b/drivers/gpu/drm/msm/disp/mdp5/mdp5_cfg.c
index 824067d2d427..05ad159b6261 100644
--- a/drivers/gpu/drm/msm/disp/mdp5/mdp5_cfg.c
+++ b/drivers/gpu/drm/msm/disp/mdp5/mdp5_cfg.c
@@ -553,6 +553,91 @@ const struct mdp5_cfg_hw msm8x96_config = {
 	.max_clk = 412500000,
 };
 
+const struct mdp5_cfg_hw msm8917_config = {
+	.name = "msm8917",
+	.mdp = {
+		.count = 1,
+		.caps = MDP_CAP_CDM,
+	},
+	.ctl = {
+		.count = 3,
+		.base = { 0x01000, 0x01200, 0x01400 },
+		.flush_hw_mask = 0xffffffff,
+	},
+	.pipe_vig = {
+		.count = 1,
+		.base = { 0x04000 },
+		.caps = MDP_PIPE_CAP_HFLIP	|
+			MDP_PIPE_CAP_VFLIP	|
+			MDP_PIPE_CAP_SCALE	|
+			MDP_PIPE_CAP_CSC	|
+			MDP_PIPE_CAP_DECIMATION	|
+			MDP_PIPE_CAP_SW_PIX_EXT	|
+			0,
+	},
+	.pipe_rgb = {
+		.count = 2,
+		.base = { 0x14000, 0x16000 },
+		.caps = MDP_PIPE_CAP_HFLIP	|
+			MDP_PIPE_CAP_VFLIP	|
+			MDP_PIPE_CAP_DECIMATION	|
+			MDP_PIPE_CAP_SW_PIX_EXT	|
+			0,
+	},
+	.pipe_dma = {
+		.count = 1,
+		.base = { 0x24000 },
+		.caps = MDP_PIPE_CAP_HFLIP	|
+			MDP_PIPE_CAP_VFLIP	|
+			MDP_PIPE_CAP_SW_PIX_EXT	|
+			0,
+	},
+	.pipe_cursor = {
+		.count = 1,
+		.base = { 0x34000 },
+		.caps = MDP_PIPE_CAP_HFLIP	|
+			MDP_PIPE_CAP_VFLIP	|
+			MDP_PIPE_CAP_SW_PIX_EXT	|
+			MDP_PIPE_CAP_CURSOR	|
+			0,
+	},
+
+	.lm = {
+		.count = 2,
+		.base = { 0x44000, 0x45000 },
+		.instances = {
+				{ .id = 0, .pp = 0, .dspp = 0,
+				  .caps = MDP_LM_CAP_DISPLAY, },
+				{ .id = 1, .pp = -1, .dspp = -1,
+				  .caps = MDP_LM_CAP_WB },
+			     },
+		.nb_stages = 8,
+		.max_width = 2048,
+		.max_height = 0xFFFF,
+	},
+	.dspp = {
+		.count = 1,
+		.base = { 0x54000 },
+
+	},
+	.pp = {
+		.count = 1,
+		.base = { 0x70000 },
+	},
+	.cdm = {
+		.count = 1,
+		.base = { 0x79200 },
+	},
+	.intf = {
+		.base = { 0x6a000, 0x6a800 },
+		.connect = {
+			[0] = INTF_DISABLED,
+			[1] = INTF_DSI,
+		},
+	},
+	.max_clk = 320000000,
+};
+
 static const struct mdp5_cfg_handler cfg_handlers[] = {
 	{ .revision = 0, .config = { .hw = &msm8x74v1_config } },
 	{ .revision = 2, .config = { .hw = &msm8x74v2_config } },
@@ -560,6 +645,7 @@ static const struct mdp5_cfg_handler cfg_handlers[] = {
 	{ .revision = 6, .config = { .hw = &msm8x16_config } },
 	{ .revision = 9, .config = { .hw = &msm8x94_config } },
 	{ .revision = 7, .config = { .hw = &msm8x96_config } },
+	{ .revision = 15, .config = { .hw = &msm8917_config } },
 };
 
 static struct mdp5_cfg_platform *mdp5_get_config(struct platform_device *dev);
